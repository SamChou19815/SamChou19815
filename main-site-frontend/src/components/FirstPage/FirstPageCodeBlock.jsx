// @flow strict

import type { Node } from 'react';
import React from 'react';
import CodeBlock from '../Common/CodeBlock';
import styles from './FirstPageCodeBlock.module.css';

const code = `

/**
 * Copyright (C) 2015â€“2019 Developer Sam.
 *
 * @author sam
 */

// Basic information of sam, written in SAMLANG.
// The code below is a well-typed SAMLANG program.
// Try it by yourself at
// https://samlang-demo.developersam.com
 
class List<T>(Nil(unit), Cons([T * List<T>])) {
  public function <T> of(t: T): List<T> =
    Cons([t, Nil(unit)])
  public method cons(t: T): List<T> =
    Cons([t, this])
}
class Developer(
  name: string, github: string,
  projects: List<string>,
) {
  public function sam(): Developer =
    val l = List::of("SAMLANG")::cons("...");
    val github = "SamChou19815";
    { name: "Sam Zhou", github, projects: l }
}
util Main {
  function main(): Developer = Developer::sam()
}
`;

export default (): Node => <CodeBlock className={styles.Block}>{code}</CodeBlock>;
