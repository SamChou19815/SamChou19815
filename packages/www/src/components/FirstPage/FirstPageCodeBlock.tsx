import React, { ReactElement } from 'react';

import CodeBlock from 'lib-react/PrismCodeBlock';

import styles from './FirstPageCodeBlock.module.css';

const code = `

/**
 * Copyright (C) 2015â€“2020 Developer Sam.
 *
 * @author sam
 */

// Basic information of sam, written in SAMLANG.
// The code below is a well-typed SAMLANG program.
// Try it by yourself at
// https://samlang-demo.developersam.com

class List<T>(Nil(unit), Cons([T * List<T>])) {
  function <T> of(t: T): List<T> =
    Cons([t, Nil({})])
  method cons(t: T): List<T> =
    Cons([t, this])
}
class Developer(
  val name: string, val github: string,
  val projects: List<string>,
) {
  function sam(): Developer = {
    val l = List.of("SAMLANG").cons("...")
    val github = "SamChou19815"
    { name: "Sam Zhou", github, projects: l }
  }
}
class Main {
  function main(): Developer = Developer.sam()
}
`;

export default (): ReactElement => (
  <CodeBlock language="samlang" className={styles.Block}>
    {code}
  </CodeBlock>
);
