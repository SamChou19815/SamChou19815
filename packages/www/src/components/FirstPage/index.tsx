import React, { ReactElement, Suspense, lazy } from 'react';
import Card from '@material-ui/core/Card';
import CardHeader from '@material-ui/core/CardHeader';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import AccountCirlce from '@material-ui/icons/AccountCircle';
import Code from '@material-ui/icons/Code';
import Domain from '@material-ui/icons/Domain';
import School from '@material-ui/icons/School';
import Work from '@material-ui/icons/Work';
import MaterialButtonLink from 'lib-react/MaterialButtonLink';
import { SvgIconProps } from '@material-ui/core/SvgIcon';
import styles from './index.module.css';

const FirstPageCodeBlock = lazy(() => import('./FirstPageCodeBlock'));

type IconLineProps = {
  readonly Icon: (props: SvgIconProps) => ReactElement;
  readonly children: string;
};

const IconLine = ({ Icon, children }: IconLineProps): ReactElement => (
  <CardContent className={styles.IconLine}>
    <Icon />
    <span className={styles.IconLineText}>{children}</span>
  </CardContent>
);

export default (): ReactElement => (
  <div id="about" className={styles.FirstPage}>
    <Suspense fallback={<div style={{ width: '430px', height: '660px' }}>Loading...</div>}>
      <FirstPageCodeBlock />
    </Suspense>
    <Card className={styles.InfoCard}>
      <CardHeader avatar={<AccountCirlce />} title="Sam Zhou" subheader="About Myself" />
      <IconLine Icon={Work}>Cornell DTI Dev Lead</IconLine>
      <IconLine Icon={Work}>Facebook SWE Intern</IconLine>
      <IconLine Icon={School}>Cornell University</IconLine>
      <IconLine Icon={Domain}>Computer Science</IconLine>
      <IconLine Icon={Code}>Coding since 13</IconLine>
      <CardActions>
        <MaterialButtonLink href="#projects">Projects</MaterialButtonLink>
        <MaterialButtonLink href="#tech-talks">Talks</MaterialButtonLink>
        <MaterialButtonLink href="#timeline">Timeline</MaterialButtonLink>
      </CardActions>
    </Card>
  </div>
);
