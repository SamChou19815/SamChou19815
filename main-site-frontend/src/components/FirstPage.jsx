// @flow strict

import type { Node } from 'react';
import React from 'react';
// $FlowFixMe
import Card from '@material-ui/core/Card';
// $FlowFixMe
import CardHeader from '@material-ui/core/CardHeader';
// $FlowFixMe
import CardContent from '@material-ui/core/CardContent';
// $FlowFixMe
import Icon from '@material-ui/core/Icon';
import styles from './FirstPage.module.css';
import FirstPageCodeBlock from './FirstPageCodeBlock';

const IconLine = ({ icon, children }: {| +icon: string; +children: string |}): Node => (
  <CardContent className={styles.IconLine}>
    <Icon>{icon}</Icon>
    <span className={styles.IconLineText}>{children}</span>
  </CardContent>
);

export default (): Node => (
  <div className={styles.FirstPage}>
    <FirstPageCodeBlock />
    <Card className={styles.InfoCard}>
      <CardHeader title="Sam Zhou" />
      <IconLine icon="work">Facebook SWE Intern</IconLine>
      <IconLine icon="work">Cornell DTI TPM</IconLine>
      <IconLine icon="school">Cornell University</IconLine>
      <IconLine icon="domain">Computer Science</IconLine>
      <IconLine icon="bar_chart">GPA: 4.18/4.3</IconLine>
      <IconLine icon="code">Coding since 13</IconLine>
    </Card>
  </div>
);
