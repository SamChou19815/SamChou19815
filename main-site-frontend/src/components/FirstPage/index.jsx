// @flow strict

import type { Node } from 'react';
import React from 'react';
// $FlowFixMe
import Card from '@material-ui/core/Card';
// $FlowFixMe
import CardHeader from '@material-ui/core/CardHeader';
// $FlowFixMe
import CardActions from '@material-ui/core/CardActions';
// $FlowFixMe
import CardContent from '@material-ui/core/CardContent';
// $FlowFixMe
import Icon from '@material-ui/core/Icon';
import styles from './FirstPage.module.css';
import FirstPageCodeBlock from './FirstPageCodeBlock';
import ButtonLink from '../Common/ButtonLink';

const IconLine = ({ icon, children }: {| +icon: string; +children: string |}): Node => (
  <CardContent className={styles.IconLine}>
    <Icon>{icon}</Icon>
    <span className={styles.IconLineText}>{children}</span>
  </CardContent>
);

export default (): Node => (
  <div id="about" className={styles.FirstPage}>
    <FirstPageCodeBlock />
    <Card className={styles.InfoCard}>
      <CardHeader avatar={<Icon>account_circle</Icon>} title="Sam Zhou" subheader="About Myself" />
      <IconLine icon="work">Facebook SWE Intern</IconLine>
      <IconLine icon="work">Cornell DTI TPM</IconLine>
      <IconLine icon="school">Cornell University</IconLine>
      <IconLine icon="domain">Computer Science</IconLine>
      <IconLine icon="bar_chart">GPA: 4.18/4.3</IconLine>
      <IconLine icon="code">Coding since 13</IconLine>
      <CardActions>
        <ButtonLink href="#projects">Projects</ButtonLink>
        <ButtonLink href="#timeline">Timeline</ButtonLink>
      </CardActions>
    </Card>
  </div>
);
