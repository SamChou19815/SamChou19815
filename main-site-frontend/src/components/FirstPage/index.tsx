import React, { ReactElement } from 'react';
import Card from '@material-ui/core/Card';
import CardHeader from '@material-ui/core/CardHeader';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import Icon from '@material-ui/core/Icon';
import styles from './index.module.css';
import FirstPageCodeBlock from './FirstPageCodeBlock';
import ButtonLink from '../Common/ButtonLink';

type IconLineProps = {
  readonly icon: string;
  readonly children: string;
}

const IconLine = ({ icon, children }: IconLineProps): ReactElement => (
  <CardContent className={styles.IconLine}>
    <Icon>{icon}</Icon>
    <span className={styles.IconLineText}>{children}</span>
  </CardContent>
);

export default (): ReactElement => (
  <div id="about" className={styles.FirstPage}>
    <FirstPageCodeBlock />
    <Card className={styles.InfoCard}>
      <CardHeader avatar={<Icon>account_circle</Icon>} title="Sam Zhou" subheader="About Myself" />
      <IconLine icon="work">Facebook SWE Intern</IconLine>
      <IconLine icon="work">Cornell DTI Dev Lead</IconLine>
      <IconLine icon="school">Cornell University</IconLine>
      <IconLine icon="domain">Computer Science</IconLine>
      <IconLine icon="bar_chart">GPA: 4.18/4.3</IconLine>
      <IconLine icon="code">Coding since 13</IconLine>
      <CardActions>
        <ButtonLink href="#projects">Projects</ButtonLink>
        <ButtonLink href="#tech-talks">Talks</ButtonLink>
        <ButtonLink href="#timeline">Timeline</ButtonLink>
      </CardActions>
    </Card>
  </div>
);
