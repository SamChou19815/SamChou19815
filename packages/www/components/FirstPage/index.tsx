import React, { ReactElement } from 'react';

import InformationCard from '../Common/InformationCard';
import FirstPageCodeBlock from './FirstPageCodeBlock';
import styles from './index.module.css';

export default (): ReactElement => (
  <div id="about" className={styles.FirstPage}>
    <FirstPageCodeBlock />
    <InformationCard />
  </div>
);
