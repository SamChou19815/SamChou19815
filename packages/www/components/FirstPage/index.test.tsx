import React from 'react';

import renderer from 'react-test-renderer';

import FirstPage from '.';

it('FirstPage matches snapshot.', () => {
  const tree = renderer.create(<FirstPage />).toJSON();
  expect(tree).toMatchSnapshot();
});
