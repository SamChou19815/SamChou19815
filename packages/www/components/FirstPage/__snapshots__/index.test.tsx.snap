// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FirstPage matches snapshot. 1`] = `
<div
  className="FirstPage"
  id="about"
>
  <pre
    className="prism-code language-samlang Block"
    style={
      Object {
        "backgroundColor": "#f7f7f7",
      }
    }
  >
    <div
      className="token-line"
      style={
        Object {
          "backgroundColor": null,
        }
      }
    >
      <span
        className="token comment"
        style={
          Object {
            "color": "rgb(128, 128, 128)",
            "fontStyle": "italic",
          }
        }
      >
        /**
      </span>
    </div>
    <div
      className="token-line"
      style={
        Object {
          "backgroundColor": null,
        }
      }
    >
      <span
        className="token comment"
        style={
          Object {
            "color": "rgb(128, 128, 128)",
            "fontStyle": "italic",
          }
        }
      >
         * Copyright (C) 2015â€“2020 Developer Sam.
      </span>
    </div>
    <div
      className="token-line"
      style={
        Object {
          "backgroundColor": null,
        }
      }
    >
      <span
        className="token comment"
        style={
          Object {
            "color": "rgb(128, 128, 128)",
            "fontStyle": "italic",
          }
        }
      >
         *
      </span>
    </div>
    <div
      className="token-line"
      style={
        Object {
          "backgroundColor": null,
        }
      }
    >
      <span
        className="token comment"
        style={
          Object {
            "color": "rgb(128, 128, 128)",
            "fontStyle": "italic",
          }
        }
      >
         * @author sam
      </span>
    </div>
    <div
      className="token-line"
      style={
        Object {
          "backgroundColor": null,
        }
      }
    >
      <span
        className="token comment"
        style={
          Object {
            "color": "rgb(128, 128, 128)",
            "fontStyle": "italic",
          }
        }
      >
         */
      </span>
      <span
        className="token plain"
      >
        
      </span>
    </div>
    <div
      className="token-line"
      style={
        Object {
          "backgroundColor": null,
        }
      }
    >
      <span
        className="token plain"
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        
      </span>
    </div>
    <div
      className="token-line"
      style={
        Object {
          "backgroundColor": null,
        }
      }
    >
      <span
        className="token plain"
      >
        
      </span>
      <span
        className="token comment"
        style={
          Object {
            "color": "rgb(128, 128, 128)",
            "fontStyle": "italic",
          }
        }
      >
        // Basic information of sam, written in SAMLANG.
      </span>
      <span
        className="token plain"
      >
        
      </span>
    </div>
    <div
      className="token-line"
      style={
        Object {
          "backgroundColor": null,
        }
      }
    >
      <span
        className="token plain"
      >
        
      </span>
      <span
        className="token comment"
        style={
          Object {
            "color": "rgb(128, 128, 128)",
            "fontStyle": "italic",
          }
        }
      >
        // The code below is a well-typed SAMLANG program.
      </span>
      <span
        className="token plain"
      >
        
      </span>
    </div>
    <div
      className="token-line"
      style={
        Object {
          "backgroundColor": null,
        }
      }
    >
      <span
        className="token plain"
      >
        
      </span>
      <span
        className="token comment"
        style={
          Object {
            "color": "rgb(128, 128, 128)",
            "fontStyle": "italic",
          }
        }
      >
        // Try it by yourself at
      </span>
      <span
        className="token plain"
      >
        
      </span>
    </div>
    <div
      className="token-line"
      style={
        Object {
          "backgroundColor": null,
        }
      }
    >
      <span
        className="token plain"
      >
        
      </span>
      <span
        className="token comment"
        style={
          Object {
            "color": "rgb(128, 128, 128)",
            "fontStyle": "italic",
          }
        }
      >
        // https://samlang-demo.developersam.com
      </span>
      <span
        className="token plain"
      >
        
      </span>
    </div>
    <div
      className="token-line"
      style={
        Object {
          "backgroundColor": null,
        }
      }
    >
      <span
        className="token plain"
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        
      </span>
    </div>
    <div
      className="token-line"
      style={
        Object {
          "backgroundColor": null,
        }
      }
    >
      <span
        className="token plain"
      >
        
      </span>
      <span
        className="token keyword"
        style={
          Object {
            "color": "rgb(62, 122, 226)",
          }
        }
      >
        class
      </span>
      <span
        className="token plain"
      >
         
      </span>
      <span
        className="token class-name"
        style={
          Object {
            "color": "rgb(154, 48, 173)",
          }
        }
      >
        List
      </span>
      <span
        className="token plain"
      >
        &lt;
      </span>
      <span
        className="token class-name"
        style={
          Object {
            "color": "rgb(154, 48, 173)",
          }
        }
      >
        T
      </span>
      <span
        className="token plain"
      >
        &gt;(
      </span>
      <span
        className="token class-name"
        style={
          Object {
            "color": "rgb(154, 48, 173)",
          }
        }
      >
        Nil
      </span>
      <span
        className="token plain"
      >
        (
      </span>
      <span
        className="token keyword"
        style={
          Object {
            "color": "rgb(62, 122, 226)",
          }
        }
      >
        unit
      </span>
      <span
        className="token plain"
      >
        ), 
      </span>
      <span
        className="token class-name"
        style={
          Object {
            "color": "rgb(154, 48, 173)",
          }
        }
      >
        Cons
      </span>
      <span
        className="token plain"
      >
        ([
      </span>
      <span
        className="token class-name"
        style={
          Object {
            "color": "rgb(154, 48, 173)",
          }
        }
      >
        T
      </span>
      <span
        className="token plain"
      >
         
      </span>
      <span
        className="token operator"
        style={
          Object {
            "color": "rgb(56, 72, 79)",
          }
        }
      >
        *
      </span>
      <span
        className="token plain"
      >
         
      </span>
      <span
        className="token class-name"
        style={
          Object {
            "color": "rgb(154, 48, 173)",
          }
        }
      >
        List
      </span>
      <span
        className="token plain"
      >
        &lt;
      </span>
      <span
        className="token class-name"
        style={
          Object {
            "color": "rgb(154, 48, 173)",
          }
        }
      >
        T
      </span>
      <span
        className="token plain"
      >
        &gt;])) {
      </span>
    </div>
    <div
      className="token-line"
      style={
        Object {
          "backgroundColor": null,
        }
      }
    >
      <span
        className="token plain"
      >
          
      </span>
      <span
        className="token keyword"
        style={
          Object {
            "color": "rgb(62, 122, 226)",
          }
        }
      >
        function
      </span>
      <span
        className="token plain"
      >
         &lt;
      </span>
      <span
        className="token class-name"
        style={
          Object {
            "color": "rgb(154, 48, 173)",
          }
        }
      >
        T
      </span>
      <span
        className="token plain"
      >
        &gt; 
      </span>
      <span
        className="token function"
        style={
          Object {
            "color": "rgb(213, 34, 98)",
          }
        }
      >
        of
      </span>
      <span
        className="token plain"
      >
        (t: 
      </span>
      <span
        className="token class-name"
        style={
          Object {
            "color": "rgb(154, 48, 173)",
          }
        }
      >
        T
      </span>
      <span
        className="token plain"
      >
        ): 
      </span>
      <span
        className="token class-name"
        style={
          Object {
            "color": "rgb(154, 48, 173)",
          }
        }
      >
        List
      </span>
      <span
        className="token plain"
      >
        &lt;
      </span>
      <span
        className="token class-name"
        style={
          Object {
            "color": "rgb(154, 48, 173)",
          }
        }
      >
        T
      </span>
      <span
        className="token plain"
      >
        &gt; =
      </span>
    </div>
    <div
      className="token-line"
      style={
        Object {
          "backgroundColor": null,
        }
      }
    >
      <span
        className="token plain"
      >
            
      </span>
      <span
        className="token class-name"
        style={
          Object {
            "color": "rgb(154, 48, 173)",
          }
        }
      >
        Cons
      </span>
      <span
        className="token plain"
      >
        ([t, 
      </span>
      <span
        className="token class-name"
        style={
          Object {
            "color": "rgb(154, 48, 173)",
          }
        }
      >
        Nil
      </span>
      <span
        className="token plain"
      >
        ({})])
      </span>
    </div>
    <div
      className="token-line"
      style={
        Object {
          "backgroundColor": null,
        }
      }
    >
      <span
        className="token plain"
      >
          
      </span>
      <span
        className="token keyword"
        style={
          Object {
            "color": "rgb(62, 122, 226)",
          }
        }
      >
        method
      </span>
      <span
        className="token plain"
      >
         
      </span>
      <span
        className="token function"
        style={
          Object {
            "color": "rgb(213, 34, 98)",
          }
        }
      >
        cons
      </span>
      <span
        className="token plain"
      >
        (t: 
      </span>
      <span
        className="token class-name"
        style={
          Object {
            "color": "rgb(154, 48, 173)",
          }
        }
      >
        T
      </span>
      <span
        className="token plain"
      >
        ): 
      </span>
      <span
        className="token class-name"
        style={
          Object {
            "color": "rgb(154, 48, 173)",
          }
        }
      >
        List
      </span>
      <span
        className="token plain"
      >
        &lt;
      </span>
      <span
        className="token class-name"
        style={
          Object {
            "color": "rgb(154, 48, 173)",
          }
        }
      >
        T
      </span>
      <span
        className="token plain"
      >
        &gt; =
      </span>
    </div>
    <div
      className="token-line"
      style={
        Object {
          "backgroundColor": null,
        }
      }
    >
      <span
        className="token plain"
      >
            
      </span>
      <span
        className="token class-name"
        style={
          Object {
            "color": "rgb(154, 48, 173)",
          }
        }
      >
        Cons
      </span>
      <span
        className="token plain"
      >
        ([t, 
      </span>
      <span
        className="token keyword"
        style={
          Object {
            "color": "rgb(62, 122, 226)",
          }
        }
      >
        this
      </span>
      <span
        className="token plain"
      >
        ])
      </span>
    </div>
    <div
      className="token-line"
      style={
        Object {
          "backgroundColor": null,
        }
      }
    >
      <span
        className="token plain"
      >
        }
      </span>
    </div>
    <div
      className="token-line"
      style={
        Object {
          "backgroundColor": null,
        }
      }
    >
      <span
        className="token plain"
      >
        
      </span>
      <span
        className="token keyword"
        style={
          Object {
            "color": "rgb(62, 122, 226)",
          }
        }
      >
        class
      </span>
      <span
        className="token plain"
      >
         
      </span>
      <span
        className="token class-name"
        style={
          Object {
            "color": "rgb(154, 48, 173)",
          }
        }
      >
        Developer
      </span>
      <span
        className="token plain"
      >
        (
      </span>
    </div>
    <div
      className="token-line"
      style={
        Object {
          "backgroundColor": null,
        }
      }
    >
      <span
        className="token plain"
      >
          
      </span>
      <span
        className="token keyword"
        style={
          Object {
            "color": "rgb(62, 122, 226)",
          }
        }
      >
        val
      </span>
      <span
        className="token plain"
      >
         name: 
      </span>
      <span
        className="token keyword"
        style={
          Object {
            "color": "rgb(62, 122, 226)",
          }
        }
      >
        string
      </span>
      <span
        className="token plain"
      >
        , 
      </span>
      <span
        className="token keyword"
        style={
          Object {
            "color": "rgb(62, 122, 226)",
          }
        }
      >
        val
      </span>
      <span
        className="token plain"
      >
         github: 
      </span>
      <span
        className="token keyword"
        style={
          Object {
            "color": "rgb(62, 122, 226)",
          }
        }
      >
        string
      </span>
      <span
        className="token plain"
      >
        ,
      </span>
    </div>
    <div
      className="token-line"
      style={
        Object {
          "backgroundColor": null,
        }
      }
    >
      <span
        className="token plain"
      >
          
      </span>
      <span
        className="token keyword"
        style={
          Object {
            "color": "rgb(62, 122, 226)",
          }
        }
      >
        val
      </span>
      <span
        className="token plain"
      >
         projects: 
      </span>
      <span
        className="token class-name"
        style={
          Object {
            "color": "rgb(154, 48, 173)",
          }
        }
      >
        List
      </span>
      <span
        className="token plain"
      >
        &lt;
      </span>
      <span
        className="token keyword"
        style={
          Object {
            "color": "rgb(62, 122, 226)",
          }
        }
      >
        string
      </span>
      <span
        className="token plain"
      >
        &gt;,
      </span>
    </div>
    <div
      className="token-line"
      style={
        Object {
          "backgroundColor": null,
        }
      }
    >
      <span
        className="token plain"
      >
        ) {
      </span>
    </div>
    <div
      className="token-line"
      style={
        Object {
          "backgroundColor": null,
        }
      }
    >
      <span
        className="token plain"
      >
          
      </span>
      <span
        className="token keyword"
        style={
          Object {
            "color": "rgb(62, 122, 226)",
          }
        }
      >
        function
      </span>
      <span
        className="token plain"
      >
         
      </span>
      <span
        className="token function"
        style={
          Object {
            "color": "rgb(213, 34, 98)",
          }
        }
      >
        sam
      </span>
      <span
        className="token plain"
      >
        (): 
      </span>
      <span
        className="token class-name"
        style={
          Object {
            "color": "rgb(154, 48, 173)",
          }
        }
      >
        Developer
      </span>
      <span
        className="token plain"
      >
         = {
      </span>
    </div>
    <div
      className="token-line"
      style={
        Object {
          "backgroundColor": null,
        }
      }
    >
      <span
        className="token plain"
      >
            
      </span>
      <span
        className="token keyword"
        style={
          Object {
            "color": "rgb(62, 122, 226)",
          }
        }
      >
        val
      </span>
      <span
        className="token plain"
      >
         l = 
      </span>
      <span
        className="token class-name"
        style={
          Object {
            "color": "rgb(154, 48, 173)",
          }
        }
      >
        List
      </span>
      <span
        className="token plain"
      >
        .
      </span>
      <span
        className="token function"
        style={
          Object {
            "color": "rgb(213, 34, 98)",
          }
        }
      >
        of
      </span>
      <span
        className="token plain"
      >
        (
      </span>
      <span
        className="token string"
        style={
          Object {
            "color": "rgb(26, 143, 82)",
          }
        }
      >
        "SAMLANG"
      </span>
      <span
        className="token plain"
      >
        ).
      </span>
      <span
        className="token function"
        style={
          Object {
            "color": "rgb(213, 34, 98)",
          }
        }
      >
        cons
      </span>
      <span
        className="token plain"
      >
        (
      </span>
      <span
        className="token string"
        style={
          Object {
            "color": "rgb(26, 143, 82)",
          }
        }
      >
        "..."
      </span>
      <span
        className="token plain"
      >
        )
      </span>
    </div>
    <div
      className="token-line"
      style={
        Object {
          "backgroundColor": null,
        }
      }
    >
      <span
        className="token plain"
      >
            
      </span>
      <span
        className="token keyword"
        style={
          Object {
            "color": "rgb(62, 122, 226)",
          }
        }
      >
        val
      </span>
      <span
        className="token plain"
      >
         github = 
      </span>
      <span
        className="token string"
        style={
          Object {
            "color": "rgb(26, 143, 82)",
          }
        }
      >
        "SamChou19815"
      </span>
      <span
        className="token plain"
      >
        
      </span>
    </div>
    <div
      className="token-line"
      style={
        Object {
          "backgroundColor": null,
        }
      }
    >
      <span
        className="token plain"
      >
            { name: 
      </span>
      <span
        className="token string"
        style={
          Object {
            "color": "rgb(26, 143, 82)",
          }
        }
      >
        "Sam Zhou"
      </span>
      <span
        className="token plain"
      >
        , github, projects: l }
      </span>
    </div>
    <div
      className="token-line"
      style={
        Object {
          "backgroundColor": null,
        }
      }
    >
      <span
        className="token plain"
      >
          }
      </span>
    </div>
    <div
      className="token-line"
      style={
        Object {
          "backgroundColor": null,
        }
      }
    >
      <span
        className="token plain"
      >
        }
      </span>
    </div>
    <div
      className="token-line"
      style={
        Object {
          "backgroundColor": null,
        }
      }
    >
      <span
        className="token plain"
      >
        
      </span>
      <span
        className="token keyword"
        style={
          Object {
            "color": "rgb(62, 122, 226)",
          }
        }
      >
        class
      </span>
      <span
        className="token plain"
      >
         
      </span>
      <span
        className="token class-name"
        style={
          Object {
            "color": "rgb(154, 48, 173)",
          }
        }
      >
        Main
      </span>
      <span
        className="token plain"
      >
         {
      </span>
    </div>
    <div
      className="token-line"
      style={
        Object {
          "backgroundColor": null,
        }
      }
    >
      <span
        className="token plain"
      >
          
      </span>
      <span
        className="token keyword"
        style={
          Object {
            "color": "rgb(62, 122, 226)",
          }
        }
      >
        function
      </span>
      <span
        className="token plain"
      >
         
      </span>
      <span
        className="token function"
        style={
          Object {
            "color": "rgb(213, 34, 98)",
          }
        }
      >
        main
      </span>
      <span
        className="token plain"
      >
        (): 
      </span>
      <span
        className="token class-name"
        style={
          Object {
            "color": "rgb(154, 48, 173)",
          }
        }
      >
        Developer
      </span>
      <span
        className="token plain"
      >
         = 
      </span>
      <span
        className="token class-name"
        style={
          Object {
            "color": "rgb(154, 48, 173)",
          }
        }
      >
        Developer
      </span>
      <span
        className="token plain"
      >
        .
      </span>
      <span
        className="token function"
        style={
          Object {
            "color": "rgb(213, 34, 98)",
          }
        }
      >
        sam
      </span>
      <span
        className="token plain"
      >
        ()
      </span>
    </div>
    <div
      className="token-line"
      style={
        Object {
          "backgroundColor": null,
        }
      }
    >
      <span
        className="token plain"
      >
        }
      </span>
    </div>
  </pre>
  <div
    className="MuiPaper-root MuiCard-root InfoCard MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      className="MuiCardHeader-root"
    >
      <div
        className="MuiCardHeader-avatar"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
          />
        </svg>
      </div>
      <div
        className="MuiCardHeader-content"
      >
        <span
          className="MuiTypography-root MuiCardHeader-title MuiTypography-body2 MuiTypography-displayBlock"
        >
          Sam Zhou
        </span>
        <span
          className="MuiTypography-root MuiCardHeader-subheader MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
        >
          About Myself
        </span>
      </div>
    </div>
    <div
      className="MuiCardContent-root IconLine"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M20 6h-4V4c0-1.11-.89-2-2-2h-4c-1.11 0-2 .89-2 2v2H4c-1.11 0-1.99.89-1.99 2L2 19c0 1.11.89 2 2 2h16c1.11 0 2-.89 2-2V8c0-1.11-.89-2-2-2zm-6 0h-4V4h4v2z"
        />
      </svg>
      <span
        className="IconLineText"
      >
        Cornell DTI Dev Lead
      </span>
    </div>
    <div
      className="MuiCardContent-root IconLine"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M20 6h-4V4c0-1.11-.89-2-2-2h-4c-1.11 0-2 .89-2 2v2H4c-1.11 0-1.99.89-1.99 2L2 19c0 1.11.89 2 2 2h16c1.11 0 2-.89 2-2V8c0-1.11-.89-2-2-2zm-6 0h-4V4h4v2z"
        />
      </svg>
      <span
        className="IconLineText"
      >
        Facebook SWE Intern
      </span>
    </div>
    <div
      className="MuiCardContent-root IconLine"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M5 13.18v4L12 21l7-3.82v-4L12 17l-7-3.82zM12 3L1 9l11 6 9-4.91V17h2V9L12 3z"
        />
      </svg>
      <span
        className="IconLineText"
      >
        Cornell University
      </span>
    </div>
    <div
      className="MuiCardContent-root IconLine"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M24 14V6H4v36h40V14H24zM12 38H8v-4h4v4zm0-8H8v-4h4v4zm0-8H8v-4h4v4zm0-8H8v-4h4v4zm8 24h-4v-4h4v4zm0-8h-4v-4h4v4zm0-8h-4v-4h4v4zm0-8h-4v-4h4v4zm20 24H24v-4h4v-4h-4v-4h4v-4h-4v-4h16v20zm-4-16h-4v4h4v-4zm0 8h-4v4h4v-4z"
          transform="scale(0.5, 0.5)"
        />
      </svg>
      <span
        className="IconLineText"
      >
        Computer Science
      </span>
    </div>
    <div
      className="MuiCardContent-root IconLine"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M9.4 16.6L4.8 12l4.6-4.6L8 6l-6 6 6 6 1.4-1.4zm5.2 0l4.6-4.6-4.6-4.6L16 6l6 6-6 6-1.4-1.4z"
        />
      </svg>
      <span
        className="IconLineText"
      >
        Coding since 13
      </span>
    </div>
    <div
      className="MuiCardActions-root MuiCardActions-spacing"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <a
            className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
            href="/resume.pdf"
            onBlur={[Function]}
            onFocus={[Function]}
          >
            Resume
          </a>
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <a
            className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
            href="/transcript.pdf"
            onBlur={[Function]}
            onFocus={[Function]}
          >
            Transcript
          </a>
        </span>
      </button>
    </div>
  </div>
</div>
`;
